### PR TITLE
Added typings as described in readme

### LINKED ISSUE #11: Add typings
Very nice library! May it be possible to add TypeScript typings into this project or to `@types/socket.io-file-client` to have autocomplete and type checking?

### PR DESCRIPTION
**Pull request by MickL**
I followed the description of the readme where you already added the types and return values. Anyway could you please check if this is complete? Also I was not sure about:

- `constructor` accepts options. But what options can it have?
- For `fileinfo` I did kind of guessing by looking at the code

Further, I updated the readme: Use arrow functions in the example, removed the duplicate example, added TypeScript import

Feel free to directly edit the PR or make code suggestions.

Closes #11

### CODE REVIEW

**Comment by rico345100** [comment]
Thanks for improving this module better! I'll check your code as soon as I get back home.

**Comment by MickL** [comment]
The import I wrote in the readme is not working correctly in TypeScript right now. I will fix in the next hours.

Also I noticed `socket.io-file` also needs types :)

**Comment by rico345100** [comment]
Okay, I'll check it then later.

File: readme.md [hunk 1]
```diff
@@ -53,23 +58,23 @@ var socket = io('http://localhost:3000');
 var uploader = new SocketIOFileClient(socket);
 var form = document.getElementById('form');
 
-uploader.on('start', function(fileInfo) {
+uploader.on('start', (fileInfo) => {
```
Comment (tuxpoldo) [inline readme.md:61]: This breaks the compatibility with IE11

Comment (MickL) [reply readme.md:61]: This is modern JS. Compatibility should be archived, if needed, by compilation with Webpack, Gulp, Babel, etc.

**Comment by MickL** [comment]
@rico345100 Can this be merged as it only adds interfaces and updated the readme?

File: readme.md (hunk 1 above)
Comment (rico345100) [reply readme.md:61]: @MickL Well, he is correct. This module is designed to support UMD, and also supports the old JavaScript environment which loads the script via <script> tag.

Comment (MickL) [reply readme.md:61]: I will revert this changes then. Are the interfaces correct tho?

**Comment by MickL** [comment]
Closed in favor of #13

**Pull request closed without merging**